### PR TITLE
fix: buffer

### PR DESCRIPTION
**Pull request by peter-jerry-ye**
- Buffer should be ignorant of the actual charset (and it may contain stuff other than string)
- Buffer should clear the content

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved accuracy of buffer length reporting.
	- Enhanced buffer reset functionality for better performance.

### CODE REVIEW

**Comment by hackwaly** [comment]
For now, our Buffer is just StringBuilder. We consider to refactor it's implementation to make it use platform specific optimizations. For example, on js target we may just use string concatenation.

In the future, we may provide a generic byte buffer. But I think use bytes directly is always an option.

**Comment by peter-jerry-ye** [comment]
> For now, our Buffer is just StringBuilder. We consider to refactor it's implementation to make it use platform specific optimizations. For example, on js target we may just use string concatenation.

We've used it to concat strings efficiently, but we have never, and should not treat it as 'StringBuilder', especially for the `length` method:
- `op_get` and `op_set` works with `Byte` rather than `Char`
- having `set_utf8_char` will make `len / 2` invalid
- the `length` has always been the number of bytes in the past (what would have been the `length / 2` for a utf-8 string?)

> In the future, we may provide a generic byte buffer. But I think use bytes directly is always an option.

It should be a `CharBuffer` #195

**Comment by peter-jerry-ye** [comment]
The `Buffer` should be considered equivalent to `ArrayBuffer` in JavaScript.

**Comment by hackwaly** [comment]
Agree to rename to respect it's purpose

**Comment by bobzhang** [comment]
let's not expose this function if it is not used

File: builtin/buffer.mbt [hunk 1]
```diff
@@ -111,15 +111,14 @@ pub fn write_char(self : Buffer, value : Char) -> Unit {
   self.len = self.len + inc
 }
 
-/// Length of containing data.
-pub fn length(self : Buffer) -> Int {
-  self.len / 2
-}
-
 pub fn reset(self : Buffer) -> Unit {
+  self.bytes = Bytes::make(1, 0)
```
[the remark this answers is not in the document]
Comment (bobzhang) [reply builtin/buffer.mbt:115]: the comments make sense, checkout the Buffer.reset in OCaml stdlib

**Comment by peter-jerry-ye** [comment]
The reset seems uncessary, I'll close this PR.

**Pull request closed without merging**